suite/src/__tests__/fast/composedb-model-sync.test.ts

```diff
@@ -109,4 +124,79 @@ describe('Sync Model and ModelInstanceDocument using ComposeDB GraphQL API', ()
     const queryResponseid = queryResponseObj?.data?.node?.id
     expect(queryResponseid).toBeDefined()
   })
+
+  test('Create 20 model instance documents in a second and query them from the same node using mult query in the same node, query 20 times ina second', async () => {
+    for (let i = 0; i < transactionIterations; i++) {
+      // TODO : Add whatever mutation gitcoin was doing
+      const createDocumentMutation = `
+    mutation MultiQuerySchema($input: CreateBasicSchemaInput!) {
+      MultiQuerySchema(input: $input) {
+        document {
+          id
+          myData
+        }
+      }
+    }
+  `
+
+      // TODO : Generate 1 kb data and create 20 documents in a second
+      const createDocumentPromises = []
+      for (let i = 0; i < parallelIterations; i++) {
+        createDocumentPromises.push(
+          (async () => {
+            const createDocumentVariables = {
+              input: {
+                content: {
+                  myData: 'my data ' + Date.now() + ' ' + i,
+                },
+              },
+            }
+
+            const response = await composeClient3.executeQuery(
+              createDocumentMutation,
+              createDocumentVariables,
+            )
+            const responseObject = await JSON.parse(JSON.stringify(response))
+            const documentId = responseObject?.data?.createBasicSchema?.document?.id
+            expect(documentId).toBeDefined()
+            return documentId
+          })(),
+        )
+      }
+
+      const documentIds = await Promise.all(createDocumentPromises)
+      documentIds.forEach((id) => expect(id).toBeDefined())
+
+      // TODO: Create a multi query instead of simple query
+      const getDocumentsByStreamIdQuery = `
+      query MultiQuerySchemaById($ids: [ID!]!) {
+        nodes(ids: $ids) {
+          ... on MultiQuerySchema {
+            id
+            myData
+          }
+        }
+      }
+    `
+
+      const getDocumentsByStreamIdVariables = {
+        ids: documentIds,
+      }
+
+      // Generate simultaneous query load of 20
+      const queryPromises = []
+      for (let i = 0; i < parallelIterations; i++) {
+        queryPromises.push(
+          composeClient3.executeQuery(getDocumentsByStreamIdQuery, getDocumentsByStreamIdVariables),
+        )
+      }
```

Comment:
What is the expected behavior here please?
Seems like this is going to execute 20 queries using the 20 IDs of the created documents? It's likely the batching logic of the ComposeDB runtime will dedupe the calls though, so in effect it should only perform a single multiquery of 20 documents.
If the goal is to create load on the node, better use the Ceramic APIs directly I think.

Reply:
We want to perform 20 multi-query operations simultaneously. So the query of getDocumentsByStreamId will have to change.